bigquery_etl/backfill/shredder_mitigation.py

```diff
@@ -29,97 +47,259 @@ class ColumnStatus(Enum):
-def get_bigquery_type(value) -> DataTypeGroup:
-    """Return the datatype of a value, grouping similar types."""
-    date_formats = [
-        ("%Y-%m-%d", date),
-        ("%Y-%m-%d %H:%M:%S", datetime),
-        ("%Y-%m-%dT%H:%M:%S", datetime),
-        ("%Y-%m-%dT%H:%M:%SZ", datetime),
-        ("%Y-%m-%d %H:%M:%S UTC", datetime),
-        ("%H:%M:%S", time),
-    ]
-    for format, dtype in date_formats:
         try:
-            if dtype == time:
-                parsed_to_time = datetime.strptime(value, format).time()
-                if isinstance(parsed_to_time, time):
-                    return DataTypeGroup.DATE
-            else:
-                parsed_to_date = datetime.strptime(value, format)
-                if isinstance(parsed_to_date, dtype):
-                    return DataTypeGroup.DATE
-        except (ValueError, TypeError):
-            continue
+            query_job = self.client.query(
+                query=sql_text,
+                job_config=bigquery.QueryJobConfig(
+                    query_parameters=parameters,
+                    use_legacy_sql=False,
+                    dry_run=False,
+                    use_query_cache=False,
+                ),
+            )
+            query_results = query_job.result()
+        except NotFound as e:
+            raise click.ClickException(
+                f"Unable to query data for {backfill_date}. Table {self.full_table_id} not found."
+            ) from e
+        rows = [dict(row) for row in query_results]
+        return rows
+
+    def compare_current_and_previous_version(
+        self,
+        date_partition_parameter,
+    ):
+        """Generate and run a data check to compare existing and backfilled data."""
+        return NotImplemented
+
+
+def get_bigquery_type(value) -> DataTypeGroup:
+    """Find the datatype of a value, grouping similar types."""
+    if isinstance(value, dt):
+        return DataTypeGroup.DATETIME
+    try:
+        if isinstance(dt.strptime(value, "%H:%M:%S").time(), time):
+            return DataTypeGroup.TIME
+    except (ValueError, TypeError, AttributeError):
+        pass
+    try:
+        value_parsed = parser.isoparse(value.replace(" UTC", "Z").replace(" ", "T"))
+        if (
+            isinstance(value_parsed, dt)
+            and value_parsed.time() == time(0, 0)
+            and isinstance(dt.strptime(value, "%Y-%m-%d"), date)
+        ):
+            return DataTypeGroup.DATE
+        if isinstance(value_parsed, dt) and value_parsed.tzinfo is None:
+            return DataTypeGroup.DATETIME
+        if isinstance(value_parsed, dt) and value_parsed.tzinfo is not None:
+            return DataTypeGroup.TIMESTAMP
+    except (ValueError, TypeError, AttributeError):
+        pass
     if isinstance(value, time):
-        return DataTypeGroup.DATE
+        return DataTypeGroup.TIME
     if isinstance(value, date):
         return DataTypeGroup.DATE
-    elif isinstance(value, bool):
+    if isinstance(value, bool):
```

Comment:
why did you move away from `elif` to `if`? 
don't think it makes a difference since you return